### PR TITLE
syft: minor refactor and fixes in entities

### PR DESCRIPTION
**Pull request by shubham3121**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/core/adp/scalar/intermediate_gamma_scalar.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def max_lipschitz_via_explicit_search(
 
         r2_diffs = np.array(
             [
-                GammaScalar(x.min_val, x.value, x.max_val, entity=x.entity).poly  # type: ignore
+                GammaScalar(x.min_val, x.value, x.max_val, entity=x.entity, prime=x.prime).poly  # type: ignore
```
Comment (rasswanth-s) [inline packages/syft/src/syft/core/adp/scalar/intermediate_gamma_scalar.py:130]: Q: I think the input scalars are type 
```Type[PhiScalars, GammaScalars]```
https://github.com/OpenMined/PySyft/blob/c3c913e225cb9a2fbd2c04a9727efb8e367b2395/packages/syft/src/syft/core/adp/scalar/abstract/intermediate_scalar.py#L59
I think the PhiScalars do not have .prime attribute,which would raise an error, 
or the input scalar are only ```GammaScalars``` in the scenario.

Comment (shubham3121) [reply packages/syft/src/syft/core/adp/scalar/intermediate_gamma_scalar.py:130]: @rasswanth-s I think we are assuming the input scalars to be of the `GammaScalar` type, cause we're are recreating the GammaScalar from the input scalars (and prime is a required parameter to do so). Maybe we add a check and filter out `GammaScalars` before calculating `r2_diffs`.

Comment (rasswanth-s) [reply packages/syft/src/syft/core/adp/scalar/intermediate_gamma_scalar.py:130]: Ahh ok , should we throw an Error if there is  `PhiScalar` (Or we do get a PhiScalar at this point) ? I am not  very comfortable with DP part of the codebase.

**Review by rasswanth-s — APPROVED** [review]
LGTM

**Pull request merged**